### PR TITLE
#27500 Prepare Braintree module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by slavvka** [comment]
```
09:03:13 Fatal error: Class Mock_RequestInterface_a2b28d50 contains 8 abstract methods and must therefore be declared abstract or implement the remaining methods (Magento\Framework\App\RequestInterface::getModuleName, Magento\Framework\App\RequestInterface::setModuleName, Magento\Framework\App\RequestInterface::getActionName, ...) in /var/www/html/vendor/phpunit/phpunit-mock-objects/src/Generator.php(264) : eval()'d code on line 1
```

File: app/code/Magento/Braintree/Test/Unit/Gateway/Request/KountPaymentDataBuilderTest.php [hunk 1]
```diff
@@ -42,11 +43,11 @@ class KountPaymentDataBuilderTest extends \PHPUnit\Framework\TestCase
     private $paymentDOMock;
 
     /**
-     * @var SubjectReader|\PHPUnit_Framework_MockObject_MockObject
+     * @var SubjectReader|\PHPUnit\Framework\MockObject\MockObject
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Gateway/Request/KountPaymentDataBuilderTest.php]: Please import this class

File: app/code/Magento/Braintree/Test/Unit/Gateway/Request/PaymentDataBuilderTest.php [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+use Magento\Braintree\Gateway\Config\Config;
 use Magento\Braintree\Gateway\Request\PaymentDataBuilder;
 use Magento\Braintree\Gateway\SubjectReader;
 use Magento\Braintree\Observer\DataAssignObserver;
 use Magento\Payment\Gateway\Data\OrderAdapterInterface;
 use Magento\Payment\Gateway\Data\PaymentDataObjectInterface;
 use Magento\Sales\Model\Order\Payment;
-use PHPUnit_Framework_MockObject_MockObject as MockObject;
-use Magento\Braintree\Gateway\Config\Config;
+use PHPUnit\Framework\MockObject\MockObject as MockObject;
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Gateway/Request/PaymentDataBuilderTest.php]: I see in many classes you have construction like that. Could you replace it by following?
```suggestion
use PHPUnit\Framework\MockObject\MockObject;
```

File: app/code/Magento/Braintree/Test/Unit/Gateway/Request/ThreeDSecureDataBuilderTest.php [hunk 3]
```diff
@@ -44,7 +45,7 @@ class ThreeDSecureDataBuilderTest extends \PHPUnit\Framework\TestCase
     private $billingAddressMock;
 
     /**
-     * @var SubjectReader|\PHPUnit_Framework_MockObject_MockObject
+     * @var SubjectReader|\PHPUnit\Framework\MockObject\MockObject
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Gateway/Request/ThreeDSecureDataBuilderTest.php]: Please import this class

File: app/code/Magento/Braintree/Test/Unit/Gateway/Request/VoidDataBuilderTest.php [hunk 4]
```diff
@@ -32,14 +33,14 @@ class VoidDataBuilderTest extends \PHPUnit\Framework\TestCase
     private $paymentMock;
 
     /**
-     * @var SubjectReader|\PHPUnit_Framework_MockObject_MockObject
+     * @var SubjectReader|\PHPUnit\Framework\MockObject\MockObject
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Gateway/Request/VoidDataBuilderTest.php]: Please import this class

File: app/code/Magento/Braintree/Test/Unit/Gateway/Response/ThreeDSecureDetailsHandlerTest.php [hunk 5]
```diff
@@ -7,35 +7,32 @@
...
 
     /**
-     * @var \Magento\Sales\Model\Order\Payment|MockObject
+     * @var Payment|MockObject
      */
     private $paymentMock;
 
     /**
-     * @var SubjectReader|\PHPUnit_Framework_MockObject_MockObject
+     * @var SubjectReader|\PHPUnit\Framework\MockObject\MockObject
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Gateway/Response/ThreeDSecureDetailsHandlerTest.php]: Please import it

File: app/code/Magento/Braintree/Test/Unit/Model/Paypal/Helper/QuoteUpdaterTest.php [hunk 6]
```diff
@@ -203,7 +203,7 @@ private function updateShippingAddressStep(array $details): void
      * @param MockObject $address
      * @param array $addressData
      */
-    private function updateAddressDataStep(MockObject $address, array $addressData): void
+    private function updateAddressDataStep(\PHPUnit\Framework\MockObject\MockObject $address, array $addressData): void
```
Comment (ihor-sviziev) [inline app/code/Magento/Braintree/Test/Unit/Model/Paypal/Helper/QuoteUpdaterTest.php]: Could you import it?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz,
Please fix the issues I found and fatal error that @slavvka mentioned above.

**Comment by slavvka** [comment]
```
21:19:42 Fatal error: Class Mock_RequestInterface_aa8ba27b contains 8 abstract methods and must therefore be declared abstract or implement the remaining methods (Magento\Framework\App\RequestInterface::getModuleName, Magento\Framework\App\RequestInterface::setModuleName, Magento\Framework\App\RequestInterface::getActionName, ...) in /var/www/html/vendor/phpunit/phpunit-mock-objects/src/Generator.php(264) : eval()'d code on line 1
```

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved 
I did small fix on one file, but everything else looked fine and unit tests were passing on last commit.

**Pull request merged**